corehq/apps/export/models/new.py

```diff
@@ -478,6 +507,30 @@ def _merge_lists(one, two, keyfn, resolvefn, copyfn):
     return merged
 
 
+def _merge_dicts(one, two, resolvefn):
+    """Merges two dicts. The algorithm is to first create a dictionary of all the keys that exist in one and
+    two but not in both. Then iterate over each key that belongs in both while calling the resovlefn function
+    to ensure the propery value gets set.
+
+    :param one: The first dictionary
+    :param two: The second dictionary
+    :param resolvefn: A function that takes two values and resolves to one
+    :returns: The merged dictionary
+    """
+    # keys either in one or two, but not both
+    merged = {
+        key: one.get(key, two.get(key))
+        for key in one.viewkeys() ^ two.viewkeys()
+    }
+
+    # merge keys that exist in both
+    merged.update({
+        key: resolvefn(one[key], two[key])
+        for key in one.viewkeys() & two.viewkeys()
+    })
+    return merged
```

Comment:
:candy: it's tested here: https://github.com/dimagi/commcare-hq/pull/10107/files#diff-7666ae1c74ee280eef29d4b5c1a96740R140